### DISCUSSION TITLE
Не открывается карта выбора ПВЗ

### DISCUSSION THREAD

**Discussion by d-nikolaev** [cdek-it/wordpress#75] [Q&A] [answered]
При оформлении заказа, при выбор Доставка до пункта выдачи. Не работает кнопка Выберите ПВЗ.
При клике на кнопку ничего не происходит.
![ПВЗ](https://github.com/user-attachments/assets/8eef2d5d-ac6b-491e-a8a8-09433859559f)

**Comment by vermorag** [comment]
Добрый день. Могу предположить, что у вас не стандартная страница чекаута - поэтому на ней не подгружаются скрипты, которые отвечают за рендер карты.
Проверьте, что в настройках выбрана именно страница, на которой расположен чекаут.
![image](https://github.com/user-attachments/assets/eff2b272-6bf7-4fd9-a2a7-602217fba513)

Если это не поможет - можете прислать на [cms-integrator@cdek.ru](mailto:cms-integrator@cdek.ru) ссылку на вход в панель администрирования, а также логин и пароль. Во избежание путаницы, в письме также укажите ссылку на данный топик, а в теме “Предоставление доступов”.

Обращу внимание, что данная почта используется только для безопасной передачи учетных данных, техническая поддержка по ней не производится.

Сделаю персональную диагностику и отпишусь о причинах.

**Comment by vermorag** [comment]
Получил доступы, нашел в чем причина.
На вашем сайте стоит стороннее расширение, которое добавляет атрибут `defer` ко всем подгружаемым скриптам. Однако, это расширение не учитывает дополнительно выводимый код к этим скриптам.

Так, например, на странице корзины (с которой наш плагин никак не взаимодействует) есть такая картина:
![image](https://github.com/user-attachments/assets/554c2b05-874f-4419-9af2-0fed2f854719)
На 777 строке подгружается js c i18n кодом. А на 779 он конфигурируется.

Однако, из-за наличия аттрибута `defer` - загрузка скрипта на 777 строке откладывается, а вот 779 выполняется сразу же, в момент загрузки страницы. Таким образом мы получаем ошибку:
![image](https://github.com/user-attachments/assets/6fc11f61-4393-4368-8f20-68765ebf109f)

Возвращаясь к нашему плагину и странице чекаута происходит та же проблема: 
![image](https://github.com/user-attachments/assets/015183b3-a77f-45f8-a1bd-e00d0227e6fb)
Наш плагин зависит и от i18n скрипта, и от lodash скрипта. 
Для последнего, wp [добавляет](https://github.com/WordPress/WordPress/blob/master/wp-includes/script-loader.php#L138) дополнительный код, который мы и видим на скриншоте ниже.
В нормальных условиях, он выполнится после подгрузки самого lodash, но из-за наличия `defer` - выполняется раньше и настройка библиотеки не происходит. 

Соответственно, позднее, наш плагин, при нажатии на кнопку выбора пвз не может работать с библиотекой, поскольку она не была должны образом настроена со стороны wp, отчего никакого эффекта на страницу не происходит.
![image](https://github.com/user-attachments/assets/4a71d9f2-d766-4eff-b4d2-7e08e826bc57)

Для корректной работы кнопки, вам необходимо исправить настройку расширения, которое вы установили для оптимизации/повышения производительности и отключить добавления аттрибута `defer` (в некоторых расширениях это назвывается отложенной подгрузкой скриптов), хотябы для страницы оформления заказа, поскольку это меняет заложенную в wp логику, на которую опирается наш плагин при работе. 
Сделать исправление со своей стороны (вложить lodash библиотеку внутрь нашего плагина) мы не можем, поскольку это будет нарушать [правила разработки расширений для wp](https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#13-plugins-must-use-wordpress-default-libraries).

**Comment by d-nikolaev** [comment]
Добрый день.
В поиск решения "стоит стороннее расширение, которое добавляет атрибут defer" не получилось обнаружить его.
Что сделал.
Отключил все плагины за исключение некоторых самого woocommerce, яндекс касса, страница заглушки (ваш сайт на техническом обслуживании для клиентов), плагин кэширования, плагин сдэк
![cdek_4](https://github.com/user-attachments/assets/a474ed38-8977-43b7-9339-9548a109e24d)
Начал смотреть.
Смотрю и вижу что вообще теперь даже нет предложения по сдэк доставке
![cdek_1](https://github.com/user-attachments/assets/4f32c7b9-36ee-4d63-9b6d-2b5d62baf720)
![cdek_2](https://github.com/user-attachments/assets/7f52d800-5d62-4cb3-9370-4c544c16aa76)
![cdek_3](https://github.com/user-attachments/assets/0aa2faaf-ff7e-43e9-b79a-52fde3862910)
![cdek_5](https://github.com/user-attachments/assets/cfefa279-ed01-4462-b8e9-e79768b3f28d)
При этом неважно какой адрес и город.
Даже решил сменить тему оформления на базовую Storefront.
Ничего не помогло.

**Reply by vermorag** [reply]
У вас выбран ограниченные список тарифов. Если тестируете отображение методов СДЭК-а, то стоит выбирать все семейства тарифов (экспресс, ecom, посылка, экономичная посылка, etc).
Судя по логам, с параметрами последних отправленные вами запросов калькулятор возвращал тарифы семейства "Экспресс" и "Магистральный экспресс". Однако они у вас в настройках не выбраны. Поэтому метод доставки не появляется.

**Comment by d-nikolaev** [comment]
Да верно, добавил больше тарифов всё появилось.
Так же разобрался по первой части вопроса. Нашел причину всё работает.
Спасибо большое за подробные и развёрнутые ответы.

**Reply by vermorag** [reply]
Можете подсказать - какое конкретно расширение добавляло атрибут и что в его настройках необходимо переключить/отключить?

**Comment by d-nikolaev** [comment, marked as answer]
Я обнаружил что у меня были добавлены различные хуки в function.php я их убрал и всё починилось. 
Эти хуки как оказалось и не нужны были. На эту это решение навела мысль когда сменил тему оформления.

**Reply by vermorag** [reply]
Понял. Спасибо большое за ответ.
